docs/WALKTHROUGH.md

```diff
@@ -0,0 +1,394 @@
+defaults, execute this command from within the Kubernetes directory:
+
+```console
+helm install <path to service-catalog>/charts/catalog --name catalog --namespace catalog
+```
+
+Note: in the event you need to redeploy service catalog, perhaps to deploy
+different images a `helm delete --purge catalog` must be performed.
+
+## Step 2 - Understand Service Catalog Components
+
+Now that the system has been deployed to our Kubernetes cluster, we can use
+`kubectl` to talk to the service catalog API server.  The service catalog API
+has four resources:
+
+- Broker Server: A server that acts as a service broker and conforms to the 
+    [Open Service Broker API](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md)
+- `Broker` Resource: a representation of the broker server in the service-catalog system
+    that indicates where a broker server is running
+- `ServiceClass`: a service offered by a particular service broker. `ServiceClass`es are created
+    in response to a `Broker` resource being submitted to the system
+- `Instance`: an instance of a `ServiceClass` provisioned by the `Broker` for that `ServiceClass`
+- `Binding`: a binding to an `Instance` which is manifested into a Kubernetes namespace
+
+These components are building blocks of the service catalog in Kubernetes from an
+API standpoint.
+
+----
+
+#### Note: accessing the service catalog
+
+Unfortunately, `kubectl` doesn't know how to speak to both the service catalog
+API server and the main Kubernetes API server without switching contexts or
+`kubeconfig` files.  For now, the best way to access the service catalog API
+server is via a dedicated `kubeconfig` file.  You can manage the kubeconfig in
+use within a directory using the `direnv` tool.
+
+### Creating kubeconfig
+Make sure to change to the service catalog directory and create the .kubeconfig
+file copying the following into your shell:
+
+console```
+cat << EOF > .kubeconfig
+apiVersion: v1
+clusters:
+- cluster:
+    server: http://<cluster ip>:80
+  name: service-catalog-cluster
+contexts:
+- context:
+    cluster: service-catalog-cluster
+  name: service-catalog-ctx
+current-context: service-catalog-ctx
+kind: Config
+preferences: {}
+EOF
+```
+
+The above `.kubeconfig` file used in combination with `direnv` is the simplest
+way to switch between the core API server and the service-catalog API server.
+An example of an .envrc file in the service catalog directory would look like:
+
+export KUBECONFIG=<path to service-catalog>/.kubeconfig
+
+Then kubectl commands executed from the service-catalog directory will
+now make requests to the correct API server.
+
+----
+
+Because we haven't created any resources in the service-catalog API server yet,
+`kubectl get` will return an empty list of resources:
+
+```console
+kubectl get brokers,serviceclasses,instances,bindings
+```
+
+Create a test-ns namespace while still executing Kubernetes API commands:
+```console
```

Comment:
add a newline above this one. if you don't, the preformatted text won't be rendered